From 28c800ff94e3456aef2ae3f1ee30b2b40f4b622d Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Thu, 28 Mar 2024 14:35:24 +0530
Subject: [PATCH] wifi: mac80211: Add support to increase rx tx bytes for
 kernel6.1

Currently we are handling the tx_packets, rx_packets, tx_bytes,
rx_bytes for kernel version less than 6.1, for latest
kernel version increment of these values is not handled.

As for latest kernel version these values are declared as structure
in linux-6.1, incrementing values straight forward will not work
here.
Add else condition for increasing values with functions as
u64_stats_add() and u64_stats_inc() for latest Linux version.

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 net/mac80211/rx.c | 3 +++
 net/mac80211/tx.c | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index 95da08f..83e8828 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -43,6 +43,9 @@ static inline void ieee80211_rx_stats(struct net_device *dev, u32 len)
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 	tstats->rx_packets++;
 	tstats->rx_bytes += len;
+#else
+	u64_stats_inc(&tstats->rx_packets);
+	u64_stats_add(&tstats->rx_bytes, len);
 #endif
 	u64_stats_update_end(&tstats->syncp);
 }
diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index 5760e85..467ce95 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -54,6 +54,9 @@ static inline void ieee80211_tx_stats(struct net_device *dev, u32 len)
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 	tstats->tx_packets++;
 	tstats->tx_bytes += len;
+#else
+	u64_stats_inc(&tstats->tx_packets);
+	u64_stats_add(&tstats->tx_bytes, len);
 #endif
 	u64_stats_update_end(&tstats->syncp);
 }
-- 
2.34.1

